libraries/hatch-server-user-management/src/UserManagementController.ts

```diff
@@ -170,19 +190,24 @@ export default class UserManagementController {
       if (!username || !confirmationCode || !password) {
         const errMsg = 'Missing required field(s), username, confirmation code, and password are required';
         this.logger.debug(errMsg);
-        params.res.status(400).send({
+        params.res.status(HttpStatus.BAD_REQUEST).send({
           error: errMsg,
         });
       } else {
         await this.userService.confirmPasswordReset(username, confirmationCode, password);
         this.logger.debug('User password reset confirmed');
-        params.res.sendStatus(200);
+        params.res.sendStatus(HttpStatus.OK);
       }
     } catch (err) {
-      this.logger.error('Error confirming resetting user password: ', err);
-      params.res.status(500).send({
-        error: err,
-      });
+      const errorMessage = err.code ? err.code + ' - ' + err.message : err;
+      this.logger.error('Error confirming resetting user password: ' + errorMessage);
+      if (err.code === UserManagementErrorCodes.USER_NOT_FOUND ||
+        err.code === UserManagementErrorCodes.INVALID_PASSWORD ||
+        err.code === UserManagementErrorCodes.USER_NOT_CONFIRMED) {
+        params.res.sendStatus(HttpStatus.PRECONDITION_FAILED);
```

Comment:
How does the client know which of the errors was encountered? We need to be sending `code` down to the client in the response.